azext_edge/edge/providers/support/base.py

```diff
@@ -673,3 +685,181 @@ def exclude_resources_with_prefix(resources: K8sRuntimeResources, exclude_prefix
+    service_selector_functions = [
+        get_meso_resource_selectors,
+        get_meta_resource_selectors,
+        get_akri_resource_selectors,
+        get_arccontainerstorage_resource_selectors,
+        get_billing_resource_selectors,
+        get_certmanager_resource_selectors,
+        get_dataflow_resource_selectors,
+    ]
+
+    all_label_selectors = []
+    all_field_selectors = []
+
+    for get_selectors_func in service_selector_functions:
+        try:
+            resource_selectors = get_selectors_func()
+            if resource_type_str in resource_selectors:
+                selectors: ResourceSelectors = resource_selectors[resource_type_str]
+                label_selectors = selectors.get("label_selectors", [])
+                field_selectors = selectors.get("field_selectors", [])
+
+                all_label_selectors.extend(label_selectors)
+                all_field_selectors.extend(field_selectors)
+
+        except (ImportError, AttributeError) as e:
+            logger.debug(f"Skipping selector function {get_selectors_func.__name__}: {e}")
+
+    return all_label_selectors, all_field_selectors
+
+
+def _fetch_grouped_resources_by_selectors(
+    api_call: Callable[[Optional[str], Optional[str]], Any], label_selectors: List[str], field_selectors: List[str]
+) -> Optional[object]:
+    """
+    Queries Kubernetes API for resources based on provided label and field selectors.
+
+    Args:
+        api_call: A function that takes (label_selector, field_selector) and returns a resource list.
+        label_selectors: List of individual label selector strings (OR logic).
+        field_selectors: List of individual field selector strings (OR logic).
+    Returns:
+        Optional[object]: A resource list object containing all resources, or None if no resources found.
+    """
+    all_resources = []
+    resource_container = None
+
+    # Default to empty lists if no selectors provided
+    label_selectors = label_selectors or []
+    field_selectors = field_selectors or []
+
+    # If no selectors at all, return all API results
+    if not label_selectors and not field_selectors:
+        try:
+            result = api_call()
+            return result
+        except Exception as e:
+            logger.debug(f"Error fetching resources with no selectors: {e}")
+            return None
+
+    # Process all label selectors
+    for label_selector in label_selectors:
+        fetched = api_call(label_selector, None)
+        if fetched and fetched.items:
+            all_resources.extend(fetched.items)
+        # Get a container object from any successful call
+        # TODO - remove need for this hacky container / type reference
+        if fetched and not resource_container:
+            resource_container = fetched
+            resource_container.items = []
+
+    # Process all field selectors
+    for field_selector in field_selectors:
+        fetched = api_call(None, field_selector)
```

Comment:
So this process doesn't handle explicitly querying label selectors and field selectors present at the same time